### ISSUE TITLE
Ionic v4: Datetime - Can't choose dates after 2017-01-01

### ISSUE THREAD

**Issue by FdezRomero** [ionic-team/ionic#14609]
# Bug Report

**Ionic Info**
Run `ionic info` from a terminal/cmd prompt and paste the output below.

```
cli packages: (/usr/local/lib/node_modules)

   @ionic/cli-utils  : 2.0.0-rc.6
   ionic (Ionic CLI) : 4.0.0-rc.6

local packages:

   @angular-devkit/core       : 0.6.8
   @angular-devkit/schematics : 0.6.8
   @angular/cli               : 6.0.8
   @ionic/schematics-angular  : 1.0.0-rc.6
   Ionic Framework            : @ionic/angular 4.0.0-alpha.7

System:

   NodeJS : v8.11.1
   npm    : 6.1.0
   OS     : macOS High Sierra
```

**Describe the Bug**
- Default date is 2017-01-01, should be today.
- Can't choose a date after 2017-01-01, even if a `max` is defined.
- Can't accept the default value when pressing "Done", it behaves the same way as "Cancel".

**Steps to Reproduce**
Just write an `<ion-datetime>` element in any page, without any special options. It can be easily replicated in Datetime page of the v4 alpha docs.

**Related Code**
`<ion-datetime></ion-datetime>`

**Expected Behavior**
- Default date, without options, should be today.
- The current year (2018) should be able to be selected by default.
- All dates in between `min` and `max` should be able to be selected.
- The default option for each picker (index 0) should be able to be selected when pressing "Done".

**Additional Context**
Related issues:
- #14563 
- #14234

**Comment by pwespi** [comment]
> The default option for each picker (index 0) should be able to be selected when pressing "Done".

Regarding this point, see https://github.com/ionic-team/ionic/pull/14479 and https://github.com/ionic-team/ionic/pull/14480.

**Comment by liamdebeasi** [comment]
Thanks for this issue! This has been merged in by PR #17443. It will be in the next release: `4.1.0`. Please let us know if you are still seeing problems after that.
